src/btree/bt_curnext.c

```diff
@@ -280,9 +284,22 @@ __cursor_var_next(WT_CURSOR_BTREE *cbt, bool newpage, bool restart)
                 --cbt->recno;
                 continue;
             }
-            WT_RET(__wt_page_cell_data_ref(session, page, &unpack, cbt->tmp));
 
+            cbt->slot = WT_COL_SLOT(page, cip);
+            WT_RET(__wt_txn_read(session, cbt, NULL, &unpack, &upd));
+            if (upd == NULL)
+                continue;
+            if (upd != NULL && upd->type == WT_UPDATE_TOMBSTONE) {
```

Comment:
Can you add a comment explaining what situation is being handled here? I'm surprised that cursor_next is freeing updates sometimes. It would be useful having a comment describing when a tombstone is encountered in this case and why *sometimes* we free some updates.

Reply:
This is to handle the case we read a tombstone. When we see a tombstone, we should skip to the next record in cursor walk. We now return an external tombstone if the stop time pair on page is visible. In relevant history, we introduced the ext flag in WT_UPDATE indicating it is a temporary update that is not on the update chain. It is now changed to the WT_UPDATE_RESTORED_FROM_DISK flag. Here we need to free this temporary memory to avoid leaks.